sympy/printing/tests/test_str.py

```diff
@@ -328,6 +328,14 @@ def test_Poly():
 
     assert str(Poly(x - 1, x)) == "Poly(x - 1, x, domain='ZZ')"
 
+    assert str(Poly(3**(2*x), 3**x)) == "Poly((3**x)**2, (3**x), domain='ZZ')"
+    assert str(Poly((x**2)**x)) == "Poly(((x**2)**x), ((x**2)**x), domain='ZZ')"
+
+    assert str(Poly((x + y)**3, (x + y), expand=False)
+                ) == "Poly((x + y)**3, (x + y), domain='ZZ')"
```

Comment:
in that case it's return "Poly((x + y)**3, (x + y), domain='ZZ')". that's looks good.